Tested changes for ADE 1.3.2

cookeac opened the pull request:
This pull request integrates the following changes:
* Repairs to the `icarStatisticsResource` and related resources - resolves #353 
* Revert 1.3 changes to `icarConsignmentType` address fields - resolves #357 
* Added an explanation for ADE 1.3 changes around discriminators - resolves #358 
* Testing C# and Java code generation for this patch release - resolves #364

cookeac committed:
Correct icarStatisticsResource (issue 353)

1. Include "allOf" icarResource so the normal resource synchronisation etc works.
2. Move "statistics" array inside icarStatisticsGroupType - change made by @erwinspeybroeck
3. Add some more "description" values to explain the intent of the attributes.

Resolves #353

cookeac committed:
Corrected broken link in Readme.MD

And added links to the other guidance markdown files in docs.

erwinspeybroeck committed:
Update icarStatisticsResource.json

removd statistics from the required section

erwinspeybroeck committed:
Update icarStatisticsType.json

made unit as a string
and value just as number (without float and nullable)

erwinspeybroeck committed:
Create icarMetricTypeIdentifier.md

initial version

cookeac committed:
Add missing comma to fix linting error

And updated the description of the units field.

cookeac committed:
Revert address types to strings

Revert consignment originAddress and destinationAddress to simple string types, as using "anyOf" with string and PostalAddress options did not work for Java code generation.

cookeac committed:
Deprecate and replace addresses

Return `originAddress` and `destinationAddress` to string fields to avoid breaking changes in code generation, but also mark these as deprecated.
Add `originPostalAddress` and `destinationPostalAddress` as structured addresses.
Resolves #357

cookeac committed:
Add release notes

Added release notes explaining 1.3 changes and linked to them from the main readme.

cookeac commented:
This has already been reviewed by the technical working group, but our process requires a review, so linking in @erwinspeybroeck for a quick approval.

erwinspeybroeck commented:
in the milkUrlScheme:
               "operationId": "get existing milking-withdrawals",
 should be 
               "operationId": "get-existing-milking-withdrawals",

Rest is ok for me

cookeac committed:
Corrected OperationId in milk URL Scheme

in the milkUrlScheme:
"operationId": "get existing milking-withdrawals",
should be
"operationId": "get-existing-milking-withdrawals",

cookeac commented:
Correction made.
